src/app/shared/services/geoNames.js

```diff
@@ -1,14 +1,21 @@
 export default class GeoNames {
-    constructor($http, $translate) {
+    constructor($http, $ngRedux) {
         this.$http = $http;
-        this.lang = $translate.use();
         this.baseURL = 'https://secure.geonames.org';
+        this.$ngRedux = $ngRedux;
     }
 
     //maxRows -> Number of JSON objects returned (cities)
     //cities ('cities500','cities1000','cities5000','cities15000') -> Minimum quantity of population filter
     //style ('SHORT','MEDIUM','LONG','FULL') -> Quantity of data retrieved
     async getNearbyCities(city, maxRows = 7, style = 'SHORT', radius = 30, cities = 'cities500') {
+        const state = this.$ngRedux.getState();
+        if (state.main.preferedLanguage === 'gb') {
```

Comment:
this snippet is repeated several times, I think is a good idea to make this check in another place a reuse it (on the reducer, create a TranslateService, etc). Maybe is not the best solution regarding the perfomance, but this will make your code more readable and mantainable